src/sync/sync_manager.hpp

```diff
@@ -92,11 +143,20 @@ friend struct SyncSession;
     sync::Client::Reconnect m_client_reconnect_mode = sync::Client::Reconnect::normal;
     bool m_client_validate_ssl = true;
 
+    // Protects m_users
+    mutable std::mutex m_user_mutex;
```

Comment:
Is that protection necessary? Do we have multiple threads interacting with this class?

If so, can they also safely interact with the contents of the `SyncUser` objects?